frex/utils/constraint_solver.py

```diff
@@ -1,10 +1,11 @@
 from __future__ import annotations
-from ortools.linear_solver import pywraplp
-from frex.models import Candidate, ConstraintSectionSolution, ConstraintSolution
-from typing import Tuple, Optional
+from ortools.sat.python import cp_model
+from frex.models import Candidate, ConstraintSolutionSection, ConstraintSolution
+from typing import Tuple, Optional, Dict, List
 from frex.utils.common import rgetattr
-from frex.utils import ConstraintType, Constraint
+from frex.utils.constraints import ConstraintType, AttributeConstraint, SectionSetConstraint, ItemConstraint
 from enum import Enum
+from rdflib import URIRef
 
 
 class ConstraintSolver:
```

Comment:
Rather than rewrite this I would have kept the old one and had two implementations, possibly with a common superclass if it makes sense.